### PR TITLE
Fix StackTraces on Windows

### PR DESCRIPTION
**Pull request by gmittert**
Enabled backtraces on Windows. @compnerd did most of the work, but it needed a few finishing touches (making the code not ifdef'ed out :p).

Sample stack trace:
```
S:\build\Ninja-DebugAssert\swift-windows-amd64> .\bin\swiftc.exe -Xfrontend -debug-assert-immediately test.swift                                                          
Assertion failed: 0 && "This is an assertion!", file S:\swift\lib\Frontend\ArgsToFrontendOptionsConverter.cpp, line 41                                                    
Wrote crash dump file "C:\Windows\temp\dumps\swiftc.exe.9d8224.dmp"                                                                                                       
LLVMSymbolizer: error reading file: PDB Error: Unable to load PDB.  Make sure the file exists and is readable.  Calling loadDataForExe                                    
                                                                                                                                                                          
LLVMSymbolizer: error reading file: PDB Error: Unable to load PDB.  Make sure the file exists and is readable.  Calling loadDataForExe                                    
                                                                                                                                                                          
LLVMSymbolizer: error reading file: PDB Error: Unable to load PDB.  Make sure the file exists and is readable.  Calling loadDataForExe                                    
                                                                                                                                                                          
#0 0x00007ff6f5fccd4c HandleAbort s:\llvm\lib\support\windows\signals.inc:409:0                                                                                           
#1 0x00007ff94971bba1 (C:\windows\SYSTEM32\ucrtbased.dll+0x6bba1)                                                                                                         
#2 0x00007ff94971d7f9 (C:\windows\SYSTEM32\ucrtbased.dll+0x6d7f9)                                                                                                         
#3 0x00007ff949723425 (C:\windows\SYSTEM32\ucrtbased.dll+0x73425)                                                                                                         
#4 0x00007ff949722f97 (C:\windows\SYSTEM32\ucrtbased.dll+0x72f97)                                                                                                         
#5 0x00007ff949721001 (C:\windows\SYSTEM32\ucrtbased.dll+0x71001)                                                                                                         
#6 0x00007ff94972398f (C:\windows\SYSTEM32\ucrtbased.dll+0x7398f)                                                                                                         
#7 0x00007ff6ed4f7fbe debugFailWithAssertion S:\swift\lib\Frontend\ArgsToFrontendOptionsConverter.cpp:42:0                                                                
#8 0x00007ff6ed4f4f6f swift::ArgsToFrontendOptionsConverter::handleDebugCrashGroupArguments(void) S:\swift\lib\Frontend\ArgsToFrontendOptionsConverter.cpp:182:0          
#9 0x00007ff6ed4f4421 swift::ArgsToFrontendOptionsConverter::convert(class llvm::SmallVectorImpl<class std::unique_ptr<class llvm::MemoryBuffer,struct std::default_delete<class llvm::MemoryBuffer> > > *) S:\swift\lib\Frontend\ArgsToFrontendOptionsConverter.cpp:54:0                                                                           
#10 0x00007ff6ed48cefa ParseFrontendArgs S:\swift\lib\Frontend\CompilerInvocation.cpp:95:0                                                                                
#11 0x00007ff6ed48c8fb swift::CompilerInvocation::parseArgs(class llvm::ArrayRef<char const *>,class swift::DiagnosticEngine &,class llvm::SmallVectorImpl<class std::unique_ptr<class llvm::MemoryBuffer,struct std::default_delete<class llvm::MemoryBuffer> > > *,class llvm::StringRef) S:\swift\lib\Frontend\CompilerInvocation.cpp:1226:0     
#12 0x00007ff6ecc9255d swift::performFrontend(class llvm::ArrayRef<char const *>,char const *,void *,class swift::FrontendObserver *) S:\swift\lib\FrontendTool\FrontendTool.cpp:1660:0                                                                                                                                                             
#13 0x00007ff6ecb1362d run_driver S:\swift\tools\driver\driver.cpp:120:0                                                                                                  
#14 0x00007ff6ecb12856 main S:\swift\tools\driver\driver.cpp:240:0                                                                                                        
#15 0x00007ff6f60440e4 invoke_main d:\agent\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:79:0                                                           
#16 0x00007ff6f6043f8e __scrt_common_main_seh d:\agent\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288:0                                               
#17 0x00007ff6f6043e4e __scrt_common_main d:\agent\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:331:0                                                   
#18 0x00007ff6f6044179 mainCRTStartup d:\agent\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_main.cpp:17:0                                                          
#19 0x00007ff99b853034 (C:\windows\System32\KERNEL32.DLL+0x13034)                                                                                                         
#20 0x00007ff99d1b3691 (C:\windows\SYSTEM32\ntdll.dll+0x73691)                                                                                                            
<unknown>:0: error: compile command failed with exit code -2147483645 (use -v to see invocation) 
```

### CODE REVIEW

File: stdlib/public/runtime/Errors.cpp [hunk 1]
```diff
@@ -48,9 +48,11 @@
 #include <cxxabi.h>
 #endif
 
-#if SWIFT_SUPPORTS_BACKTRACE_REPORTING
-// execinfo.h is not available on Android. Checks in this file ensure that
-// fatalError behaves as expected, but without stack traces.
+#if SWIFT_SUPPORTS_BACKTRACE_REPORTING && !defined(_WIN32)
```
Comment (compnerd) [inline stdlib/public/runtime/Errors.cpp]: Please use `__has_include(<excinfo.h>)` instead

Comment (jmittert) [reply stdlib/public/runtime/Errors.cpp]: Fair enough. I removed the comment too sense it doesn't really make sense anymore.

**Comment by compnerd** [comment]
@swift-ci please test and merge

**Review by compnerd — APPROVED** [review]
Thanks for finishing this up!

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by jmittert** [comment]
> 
> 
> **Build failed**
> [Swift Test Linux Platform](https://ci.swift.org/job/swift-PR-Linux/11214/)
> Git Sha - [1abe971](https://github.com/apple/swift/commit/1abe971a16feaebe5433b97cd897114519f840d0)

lldb failed to compile?
```
19:13:28 In file included from /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/lldb/source/Plugins/ExpressionParser/Clang/ClangExpressionParser.cpp:66:
19:13:28 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/lldb/source/Plugins/ExpressionParser/Clang/ModuleDependencyCollector.h:25:46: error: only virtual member functions can be marked 'override'
19:13:28                llvm::StringRef FileDst = {}) override {
19:13:28                                              ^~~~~~~~~
19:13:28 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/lldb/source/Plugins/ExpressionParser/Clang/ModuleDependencyCollector.h:29:45: error: only virtual member functions can be marked 'override'
19:13:28   bool insertSeen(llvm::StringRef Filename) override { return false; }
19:13:28                                             ^~~~~~~~~
19:13:28 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/lldb/source/Plugins/ExpressionParser/Clang/ModuleDependencyCollector.h:30:69: error: only virtual member functions can be marked 'override'
19:13:28   void addFileMapping(llvm::StringRef VPath, llvm::StringRef RPath) override {}
19:13:28                                                                     ^~~~~~~~~
19:13:28 /home/buildnode/jenkins/workspace/swift-PR-Linux/branch-master/lldb/source/Plugins/ExpressionParser/Clang/ModuleDependencyCollector.h:31:23: error: only virtual member functions can be marked 'override'
19:13:28   void writeFileMap() override {}
19:13:28                       ^~~~~~~~~
19:13:28 4 errors generated.
```

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Comment by compnerd** [comment]
@swift-ci please test Linux platform

**Pull request merged**